### PR TITLE
Show the Create Username sheet after 1/2 sec

### PR DESCRIPTION
**Pull request by martindsq**
#803 

It presents the NIP-05 sheet from anywhere in the app half a second after the user opens the app if it didn't set the nip-05 before

### CODE REVIEW

**Comment by martindsq** [comment]
@joshuatbrown replaced the 5 sec delay with half a second

**Comment by joshuatbrown** [comment]
@martindsq that seems much better to me. Would you update the PR title and description to match?

**Review by joshuatbrown — CHANGES_REQUESTED** [review]
Some comments and questions.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Parse links and mentions in Profile's about (or bios)
 - Show “Post a reply” field at the proper height initially, and allow it to expand as more text is entered.
+- Parse links and mentions in Profile's about (or bios)
```
[the remark this answers is not in the document]
Comment (joshuatbrown) [reply CHANGELOG.md:17]: Oh I see now that we're intending to skip changelog which is fine with me. Just want to make sure this line isn't duplicated, then.

**Comment by martindsq** [comment]
@joshuatbrown I included some lines that I used for debugging or demoing the feature. I apologize by this.

I fixed that and restored the CHANGELOG file from main

File: CHANGELOG.md
Comment (joshuatbrown) [inline CHANGELOG.md:1]: Looks like we'll want to sync with `main` again -- I'm seeing a deleted line here.

File: Nos/Views/AppView.swift [hunk 2]
```diff
@@ -147,8 +148,34 @@ struct AppView: View {
...
+        // Sleep for half a second
+        try? await Task.sleep(nanoseconds: 500_000_000)
+        guard let author = currentUser.author, let npub = author.npubString else {
+            return
+        }
+        // We store the npub for the user we last presented the sheet for so that
+        // the behavior resets if the user creates a new account
+        let key = "AppView_presentedNIP05SheetFor"
+        let didPresentSheetInPast = UserDefaults.standard.string(forKey: key)
+        var shouldShowSheet: Bool
```
Comment (joshuatbrown) [inline Nos/Views/AppView.swift:168]:
```suggestion
        let shouldShowSheet: Bool
```

File: Nos/Views/AppView.swift [hunk 3]
```diff
@@ -147,8 +148,34 @@ struct AppView: View {
...
+    @Sendable private func presentNIP05SheetIfNeeded() async {
+        // Sleep for half a second
+        try? await Task.sleep(nanoseconds: 500_000_000)
+        guard let author = currentUser.author, let npub = author.npubString else {
+            return
+        }
+        // We store the npub for the user we last presented the sheet for so that
+        // the behavior resets if the user creates a new account
+        let key = "AppView_presentedNIP05SheetFor"
+        let didPresentSheetInPast = UserDefaults.standard.string(forKey: key)
```
Comment (joshuatbrown) [inline Nos/Views/AppView.swift:167]: Maybe something like this is a little more clear for me:
```suggestion
        let didPresentSheetForNpub = UserDefaults.standard.string(forKey: key)
```

File: Nos/Views/AppView.swift [hunk 4]
```diff
@@ -147,8 +148,34 @@ struct AppView: View {
...
+
+    @Sendable private func presentNIP05SheetIfNeeded() async {
+        // Sleep for half a second
+        try? await Task.sleep(nanoseconds: 500_000_000)
+        guard let author = currentUser.author, let npub = author.npubString else {
+            return
+        }
+        // We store the npub for the user we last presented the sheet for so that
+        // the behavior resets if the user creates a new account
+        let key = "AppView_presentedNIP05SheetFor"
```
Comment (joshuatbrown) [inline Nos/Views/AppView.swift:166]: I don't know how much it matters that `AppView` presented it, so I might suggest a couple tweaks:
```suggestion
        let key = "didPresentNIP05SheetForNpub"
```

File: Nos/Views/AppView.swift [hunk 5]
```diff
@@ -147,8 +148,34 @@ struct AppView: View {
...
+        let didPresentSheetInPast = UserDefaults.standard.string(forKey: key)
+        var shouldShowSheet: Bool
+        if let didPresentSheetInPast {
+            shouldShowSheet = didPresentSheetInPast != npub && author.nip05 == nil
+        } else {
+            shouldShowSheet = author.nip05 == nil
+        }
+        if shouldShowSheet {
+            showNIP05Wizard = true
+            UserDefaults.standard.setValue(npub, forKey: key)
```
Comment (joshuatbrown) [inline Nos/Views/AppView.swift:176]: Do we generally just use `UserDefaults.standard` directly like this? It's sort of a hidden dependency; at least, hidden outside of this func. Do we have a way to specify it as a `@Dependency`? I still don't fully know how all of that works.

**Review by joshuatbrown — CHANGES_REQUESTED** [review]
Looks good overall; a few implementation suggestions.

**Comment by martindsq** [comment]
@joshuatbrown It's ready to be re-reviewed!

**Review by joshuatbrown — APPROVED** [review]
Great work!

**Pull request merged**